lusidtools/cocoon/cocoon.py

```diff
@@ -734,27 +853,127 @@ def load_from_data_frame(
-    columns in the dataframe to use to create sub holdings
-    :param bool holdings_adjustment_only: Whether to use the adjust_holdings api call rather than set_holdings when
-    working with holdings
-    :param int thread_pool_max_workers: The maximum number of workers to use in the thread pool used by the function
-
-    :return: dict responses: The responses from loading the data into LUSID
+
+    Parameters
+    ----------
+    api_factory :                   lusid.utilities.ApiClientFactory api_factory
+                                    The api factory to use
+    scope :                         str
+                                    The scope of the resource to load the data into
+    data_frame :                    pd.DataFrame
+                                    The DataFrame containing the data
+    mapping_required :              dict{str, str}
+                                    The dictionary mapping the DataFrame columns to LUSID's required attributes
+    mapping_optional :              dict{str, str}
+                                    The dictionary mapping the DataFrame columns to LUSID's optional attributes
+    file_type :                     str
+                                    The type of file e.g. transactions, instruments, holdings, quotes, portfolios
+    identifier_mapping :            dict{str, str}
+                                    The dictionary mapping of LUSID instrument identifiers to identifiers in the DataFrame
+    property_columns :              list
+                                    The columns to create properties for
+    properties_scope :              str
+                                    The scope to add the properties to
+    batch_size :                    int
+                                    The size of the batch to use when using upsert calls e.g. upsert instruments, upsert quotes etc.
+    remove_white_space :            bool
+                                    remove whitespace either side of each value in the dataframe
+    instrument_name_enrichment :    bool
+                                    request additional identifier information from open-figi
+    sub_holding_keys :              str
+                                    The sub holding keys to use for this request. Can be a list of property keys or a list of
+                                    columns in the dataframe to use to create sub holdings
+    holdings_adjustment_only :      bool
+                                    Whether to use the adjust_holdings api call rather than set_holdings when working with holdings
+    thread_pool_max_workers :       int
+                                    The maximum number of workers to use in the thread pool used by the function
+
+    Returns
+    -------
+    responses:                      dict
+                                    The responses from loading the data into LUSID
```

Comment:
Why is the formatting different here with the type and description spaced out way more than the other function docstrings?

Reply:
Because this docstring was copied across from the work I was doing on this previously which was all in this format.

This format has the same semantic meaning from the docstring rendering perspective, and for this function it seemed more readable like this, however looking back on it its probably more important to be consistent 

```
def load_from_data_frame(
    api_factory: lusid.utilities.ApiClientFactory,
    scope: str,
    data_frame: pd.DataFrame,
    mapping_required: dict,
    mapping_optional: dict,
    file_type: str,
    identifier_mapping: dict = None,
    property_columns: list = None,
    properties_scope: str = None,
    batch_size: int = None,
    remove_white_space: bool = True,
    instrument_name_enrichment: bool = False,
    sub_holding_keys: list = None,
    holdings_adjustment_only: bool = False,
    thread_pool_max_workers: int = 5,
):
    """

    Parameters
    ----------
    api_factory : lusid.utilities.ApiClientFactory api_factory
        The api factory to use
    scope : str
        The scope of the resource to load the data into
    data_frame : pd.DataFrame
        The DataFrame containing the data
    mapping_required : dict{str, str}
        The dictionary mapping the DataFrame columns to LUSID's required attributes
    mapping_optional : dict{str, str}
        The dictionary mapping the DataFrame columns to LUSID's optional attributes
    file_type : str
        The type of file e.g. transactions, instruments, holdings, quotes, portfolios
    identifier_mapping : dict{str, str}
        The dictionary mapping of LUSID instrument identifiers to identifiers in the DataFrame
    property_columns : list
        The columns to create properties for
    properties_scope : str
        The scope to add the properties to
    batch_size : int
        The size of the batch to use when using upsert calls e.g. upsert instruments, upsert quotes etc.
    remove_white_space : bool
        remove whitespace either side of each value in the dataframe
    instrument_name_enrichment : bool
        request additional identifier information from open-figi
    sub_holding_keys : str
        The sub holding keys to use for this request. Can be a list of property keys or a list of
        columns in the dataframe to use to create sub holdings
    holdings_adjustment_only : bool
        Whether to use the adjust_holdings api call rather than set_holdings when working with holdings
    thread_pool_max_workers : int
        The maximum number of workers to use in the thread pool used by the function

    Returns
    -------
    responses: dict
        The responses from loading the data into LUSID
```